api/client/trust.go

```diff
@@ -377,7 +373,7 @@ func (cli *DockerCli) trustedPull(repoInfo *registry.RepositoryInfo, ref registr
 		}
 		fmt.Fprintf(cli.out, "Pull (%d of %d): %s%s@%s\n", i+1, len(refs), repoInfo.Name(), displayTag, r.digest)
 
-		if err := cli.imagePullPrivileged(authConfig, repoInfo.Name(), r.digest.String(), requestPrivilege); err != nil {
+		if err := cli.imagePullPrivileged(authConfig, repoInfo.Name()+"@"+r.digest.String(), requestPrivilege); err != nil {
```

Comment:
Do we not have a way to assemble canonical references? https://godoc.org/github.com/docker/distribution/reference#WithDigest

May have to use the docker/docker/reference version.